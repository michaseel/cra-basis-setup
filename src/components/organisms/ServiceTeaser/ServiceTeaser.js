import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const Wrapper = styled.div`
  color: red;
`;

const Headline = styled.h3`
  font-size: 20px;
`;

const ServiceTeaser = ({ href, image, title, children }) =>
  <Wrapper>
    <img src={image} alt="" />
    <Headline>{title}</Headline>
    <p>{children}</p>
  </Wrapper>;

ServiceTeaser.propTypes = {
  /** Documentation of this Prop */
  href: PropTypes.string.isRequired,
  image: PropTypes.string,
  title: PropTypes.string,
  /** Documentation of that Prop */
  children: PropTypes.string,
};

export default ServiceTeaser;
