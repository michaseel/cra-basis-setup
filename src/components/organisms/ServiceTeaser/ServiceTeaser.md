ServiceTeaser Example

    <ServiceTeaser 
      title="Garten"
      image="https://unsplash.it/300/200?image=957"
    >
      F端r einen h端bschen Garten und gesunde Pflanzen.
    </ServiceTeaser>
    
ServiceTeaser Example

    <ServiceTeaser 
      title="Garten"
      image="https://unsplash.it/300/200?image=957"
    >
      Fsdfsdfsdfsdf端r einen h端bschen Garten und gesunde Pflanzen.
    </ServiceTeaser>
    
