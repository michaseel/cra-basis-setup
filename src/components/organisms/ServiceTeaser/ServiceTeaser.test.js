import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import ServiceTeaser from './ServiceTeaser';

it('renders ServiceTeaser snapshot', () => {
  const tree = renderer
    .create(
      <ServiceTeaser title="Haus" href="#" image="https://unsplash.it/300/200?image=957">
        Für einen hübschen Garten und gesunde Pflanzen.
      </ServiceTeaser>
    )
    .toJSON();
  expect(tree).toMatchStyledComponentsSnapshot();
});
