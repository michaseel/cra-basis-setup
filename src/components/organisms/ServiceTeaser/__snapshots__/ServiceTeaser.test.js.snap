// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ServiceTeaser snapshot 1`] = `
.rOCEJ {
  color: red;
}

.hoFYxk {
  font-size: 50px;
}

<div
  className="sc-bdVaJa rOCEJ"
>
  <img
    alt=""
    src="https://unsplash.it/300/200?image=957"
  />
  <h3
    className="sc-bwzfXH hoFYxk"
    onMouseOver={[Function]}
  >
    Haus
  </h3>
  <p>
    Für einen hübschen Garten und gesunde Pflanzen.
  </p>
</div>
`;
